tempodb/encoding/vparquet5/block_autocomplete.go

```diff
@@ -14,13 +14,22 @@ import (
 	"github.com/pkg/errors"
 )
 
+type tagNameKey struct {
+	name  string
+	scope traceql.AttributeScope
+}
+
 type tagRequest struct {
 	// applies to tag names and tag values. the conditions by which to return the filtered data
 	conditions []traceql.Condition
 	// scope requested. only used for tag names. A scope of None means all scopes.
 	scope traceql.AttributeScope
 	// tag requested.  only used for tag values. if populated then return tag values for this tag, otherwise return tag names.
 	tag traceql.Attribute
+	// existsTagName is a callback to check if a tag name has already been seen
+	existsTagName func(key tagNameKey) bool
```

Comment:
I like this change of moving the cache one layer up, but is there a performance or correctness difference?  It looks like the data by each distinct attr collector doesn't overlap because they are different scopes anyway. So a map here of all data, is the same as smaller maps in the individual collectors.